packages/api/test/http.service.spec.ts

```diff
@@ -149,4 +149,58 @@ describe('HTTP Service', () => {
+
+    beforeEach(() => {
+      axiosMock.reset();
+      axiosMock.onGet('/realms/testing/.well-known/openid-configuration').reply(200, {
+        issuer: 'https://testing.hahnpro.com/auth/realms/testing',
+        grant_types_supported: ['client_credentials'],
+        token_endpoint_auth_methods_supported: ['client_secret_jwt'],
+      });
+    });
+
+    it('FLOW.HS.TOK.1 should return provided token', async () => {
+      jest
+        .useFakeTimers({
+          doNotFake: ['nextTick', 'setImmediate', 'clearImmediate', 'setInterval', 'clearInterval', 'setTimeout', 'clearTimeout'],
+        })
+        .setSystemTime(1737451350000); // A Few seconds before token expiry
+
+      const httpClient = new HttpClient('/api', 'https://testing.hahnpro.com/auth', 'testing', '', '');
+
+      await httpClient.provideExternalToken(token);
```

Comment:
das muss dann aber für jeden api call vorangestellt werden?
```ts
api.httpClient.provideExternalToken(token);
await api.assets.getOne(assetId);
```

ich fände besser:
auth header übergeben
```ts
await api.assets.getOne(assetId, { headers: authHeader });
```

oder getAccessToken irgendwie überschreiben
```ts
api.httpClient.getAccessToken = this.authService.getAccessToken;
```

oder vielleicht sogar beides

Reply:
naja das ablauf ist ja: man bekommt einen request vom user (z.B. über das Frontend) mit dem Token, man setzt den token im client und dann kann man damit dann auch mehrere api calls machen. in dem agent beispiel wir einmal z beginn der verarbeitung der user-nachricht der token gesetzt, dann machen die agenten ihre arbeit mit api calls und dann wird irgendwann wenn die nächste nachricht vom user kommt der token wieder einmal gesetzt.

Reply:
Es soll doch auch sowas ersetzt werden:

srv/services/asset-intelligence/src/ai/ai.service.ts
```ts
  const token = await this.authService.getAccessToken();
  const response: AxiosResponse<Paginated<Asset>> = await lastValueFrom(
    this.httpService.get(`${API_URL}/assets?filter=_id=@${assetIds.join(',')}`, {
      headers: { Authorization: 'Bearer ' + token },
    }),
  );
```

und man will eine instanz von API haben die injected wird

Reply:
das macht sinn, ich guck mal wie ich das beides unter einen hut bekomme